sway-core/src/build_config.rs

```diff
@@ -10,6 +13,7 @@ pub struct BuildConfig {
     pub(crate) print_intermediate_asm: bool,
     pub(crate) print_finalized_asm: bool,
     pub(crate) print_ir: bool,
+    pub(crate) generated_names: Arc<Mutex<Vec<&'static str>>>,
```

Comment:
why this instead of a `Arc<Vec<&'static str>>`?

Reply:
`Arc<Vec<&'static str>>` makes it difficult to push to the vec. There may be another way to accomplish this, but this is the one I am familiar with.